great_expectations/core/util.py

```diff
@@ -129,66 +129,70 @@ def convert_to_json_serializable(data):
+        return None
 
     if isinstance(data, (str, int, float, bool)):
         # No problem to encode json
         return data
 
-    elif isinstance(data, dict):
+    if isinstance(data, dict):
         new_dict = {}
         for key in data:
             # A pandas index can be numeric, and a dict key can be numeric, but a json key must be a string
             new_dict[str(key)] = convert_to_json_serializable(data[key])
 
         return new_dict
 
-    elif isinstance(data, (list, tuple, set)):
+    if isinstance(data, (list, tuple, set)):
         new_list = []
         for val in data:
             new_list.append(convert_to_json_serializable(val))
 
         return new_list
 
-    elif isinstance(data, (np.ndarray, pd.Index)):
+    if isinstance(data, (np.ndarray, pd.Index)):
         # test_obj[key] = test_obj[key].tolist()
         # If we have an array or index, convert it first to a list--causing coercion to float--and then round
         # to the number of digits for which the string representation will equal the float representation
         return [convert_to_json_serializable(x) for x in data.tolist()]
 
-    # Note: This clause has to come after checking for np.ndarray or we get:
-    #      `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
-    elif data is None:
-        # No problem to encode json
-        return data
-
-    elif isinstance(data, (datetime.datetime, datetime.date)):
+    if isinstance(data, (datetime.datetime, datetime.date)):
         return data.isoformat()
 
-    elif isinstance(data, uuid.UUID):
+    if isinstance(data, uuid.UUID):
         return str(data)
 
     # Use built in base type from numpy, https://docs.scipy.org/doc/numpy-1.13.0/user/basics.types.html
     # https://github.com/numpy/numpy/pull/9505
-    elif np.issubdtype(type(data), np.bool_):
+    if np.issubdtype(type(data), np.bool_):
         return bool(data)
 
-    elif np.issubdtype(type(data), np.integer) or np.issubdtype(type(data), np.uint):
+    if np.issubdtype(type(data), np.integer) or np.issubdtype(type(data), np.uint):
         return int(data)
 
-    elif np.issubdtype(type(data), np.floating):
+    if np.issubdtype(type(data), np.floating):
         # Note: Use np.floating to avoid FutureWarning from numpy
         return float(round(data, sys.float_info.dig))
 
-    elif isinstance(data, pd.Series):
+    # Note: This clause has to come after checking for np.ndarray or we get:
+    #      `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
+    if data is None:
+        # No problem to encode json
+        return data
+
+    try:
+        if not isinstance(data, list) and pd.isna(data):
+            # pd.isna is functionally vectorized, but we only want to apply this to single objects
+            # Hence, why we test for `not isinstance(list))`
+            return None
```

Comment:
Thanks for the comments here. Very helpful

Reply:
Those comments are from before (and I added another one).